cli/src/commands/log.rs

```diff
@@ -149,12 +152,24 @@ pub(crate) fn cmd_log(
         }
         expression
     };
+
+    let revset = revset_expression.evaluate()?;
+
+    if args.count {
+        let count = revset
+            .iter()
+            .take(args.limit.unwrap_or(usize::MAX))
+            .try_fold(0, |n, _| Ok::<_, RevsetEvaluationError>(n + 1))?;
```

Comment:
optional (especially since it only benefits Google right now): you could first check if `revset.count_estimate()` returns something matching `(lower, Some(upper)) if upper == lower` and not iterate over the revset in that case. That would make things like `jj log -r ..main --count` be fast at Google.

Reply:
Pushed. Is this what you had in mind?

Reply:
Yes, looks good. However, I just noticed that we use `process_results()` in the revset engine and we could use that here too. I think you can simply replace `try_fold(0, |n, _| Ok::<_, RevsetEvaluationError>(n + 1))` by `process_results(|iter| iter.count())`.